### PR TITLE
Add backward compatibility for icehouse version of keystone

### PR DESCRIPTION
**Pull request by AlvaroVega**

### CODE REVIEW

File: keystone/contrib/oauth2/backends/sql.py [hunk 1]
```diff
@@ -17,9 +17,10 @@
 from keystone.common import sql
 from keystone.contrib import oauth2
 from keystone import exception
-from keystone.i18n import _
-from oslo.utils import timeutils
-
+#from keystone.i18n import _
+from keystone.openstack.common.gettextutils import _
+#from oslo.utils import timeutils
+from keystone.openstack.common import timeutils
```
Comment (garcianavalon) [inline keystone/contrib/oauth2/backends/sql.py]: This line makes the tests fail with
`ImportError: cannot import name timeutils`

**Comment by garcianavalon** [comment]
Hi Alvaro,
running the command `sudo tox -e py27 keystone.tests.test_v3_oauth2` after applying your commits fails. The error seems to be the line `from keystone.openstack.common import timeutils`.

Is it possible to fix it or are the versions incompatible?

File: keystone/contrib/oauth2/backends/sql.py (hunk 1 above)
Comment (garcianavalon) [inline keystone/contrib/oauth2/backends/sql.py]: This line makes the test fail with a `ImportError: cannot import name timeutils`

**Comment by AlvaroVega** [comment]
You are right. Updated to allow import timeutils for icehouse or juno by catching import exception

File: keystone/contrib/oauth2/routers.py [hunk 2]
```diff
@@ -60,73 +73,132 @@ def add_routes(self, mapper):
...
+        #    })
+        mapper.connect(
+            '/users/{user_id}' + self.PATH_PREFIX + '/access_tokens/{access_token_id}',
+            controller=access_token_controller,
+            action='get_access_token',
+            conditions=dict(method=['GET']))
+        mapper.connect(
+            '/users/{user_id}' + self.PATH_PREFIX + '/access_tokens/{access_token_id}',
+            controller=access_token_controller,
+            action='delete_access_token',
```
Comment (garcianavalon) [inline keystone/contrib/oauth2/routers.py]: the controller method is called revoke_access_token not delete

File: keystone/contrib/oauth2/routers.py [hunk 3]
```diff
@@ -60,73 +73,132 @@ def add_routes(self, mapper):
...
+        #    rel=build_resource_relation(resource_name='consumers'))
+        mapper.connect(
+             self.PATH_PREFIX + '/consumers',
+             controller=consumer_controller,
+             action='create_consumer',
+             conditions=dict(method=['POST']))
+        mapper.connect(
+             self.PATH_PREFIX + '/consumers',
+             controller=consumer_controller,
+             action='list_consumer',
```
Comment (garcianavalon) [inline keystone/contrib/oauth2/routers.py]: it should be list_consumers

**Comment by garcianavalon** [comment]
There are two minor errors in routers.py. I have fixed them locally and the tests run fine now. I will do some more testing before merging though, just to be sure everything is ok.

**Comment by AlvaroVega** [comment]
Old commented code, specially in routers.py, should be removed it also

**Comment by AlvaroVega** [comment]
@garcianavalon this is ready to merge

**Comment by garcianavalon** [comment]
@AlvaroVega Thanks, I'll merge it this afternoon

**Comment by AlvaroVega** [comment]
BTW, are this repo and https://github.com/ging/keystone-oauth2-extension synchronized? The contain the same for oauth2 plugin?

**Comment by garcianavalon** [comment]
@AlvaroVega they contain the same oauth2 plugin. The idea is to remove it from here once we have a package with the extension and a way to install it automatically.

**Comment by garcianavalon** [comment]
BTW, I merged this some days ago and haven't found any issue.
Thanks for the contribution :)

**Comment by AlvaroVega** [comment]
@garcianavalon AFAIK this PR was closed, not merged.

**Pull request closed without merging**